### PR TITLE
Adds 3 new rooms and optimizes preparation scripts

### PR DESCRIPTION
**Pull request by karinnacvijanovich**
The [Motus dataset](https://ieeexplore.ieee.org/document/9610933) contains measurements obtained from 1 room with 830 unique furniture configurations and 4 source positions, resulting in a total of 3320 RIRs. The data was measured using an em32 Eigenmike microphone array, and is available as both raw Eigenmike RIRs and 25-channel 4th-order Ambisonic RIRs. 
Screenshot 2024-06-25 at 5 15 33 PM
Running inference on a pre-trained SELDnet model using data generated with SpatialScaper and the Motus dataset yields a localization error of 40.8º (mic format) and 30.2º (FOA format). 

The [Small_Conference dataset](https://pub.dega-akustik.de/DAGA_2019/data/articles/000344.pdf) contains measurements obtained from 1 room, using an mh acoustics Eigenmike placed at five unique listening positions to record 32-channel directional RIRs (DRIRs). 
Screenshot 2024-06-25 at 5 16 04 PM
Running inference on a pre-trained SELDnet model using data generated with SpatialScaper and the Small_Conference dataset yields a localization error of 43.3º (mic format). While this dataset lacks measurements in the FOA format, it's possible to convert the existing data to FOA using the same method applied to the METU dataset.

The [RSoANU dataset](https://www.mdpi.com/2076-3417/14/5/2095) contains measurements obtained from 1 room using a 32-channel spherical microphone array (mh acoustics em32 Eigenmike). Data was also collected from a B-format soundfield microphone array (Rode NT-SF1), but this FOA-formatted data appears to be spatially inconsistent. As such, the B-format data should not be integrated. Instead, the same method applied to the METU dataset could be used to convert the Eigenmike data to FOA format. Measurements were conducted for two configurations: all wall panels set to wood or felt. There were 3 source locations, and RIRs were collected over a planar receiver grid spanning the room, with the microphone array centered at a height of 1.7 m. Measrurements were also obtained from additional microphones specific to each source. There are 396 4-channel RIRs and 396 32-channel RIRs, resulting in 792 RIRs total. 
Screenshot 2024-06-25 at 5 15 10 PM
Running inference on a pre-trained SELDnet model using data generated with SpatialScaper and the RSoANU dataset yields a localization error of 33.6º (mic format).

### CODE REVIEW

**Comment by iranroman** [comment]
Thank you @karinnacvijanovich. 

As part of this PR let's revise that #67 is actually addressed (I did some testing and it seems like). 

This PR also expands the rooms, originally discussed in #6. 

It also fixes #68 issue.

File: README.md [hunk 1]
```diff
@@ -234,4 +236,31 @@ Also cite the RIR and sound event databases that SpatialScaper uses.
...
+@article{chesworth2024room,
+  title={Room Impulse Response Dataset of a Recording Studio with Variable Wall Paneling Measured Using a 32-Channel Spherical Microphone Array and a B-Format Microphone Array},
+  author={Chesworth, Grace and Bastine, Amy and Abhayapala, Thushara},
+  journal={Applied Sciences},
+  volume={14},
+  number={5},
+  pages={2095},
+  year={2024},
+  publisher={Multidisciplinary Digital Publishing Institute}
+}
```
Comment (iranroman) [inline README.md:265]: @karinnacvijanovich please add the three rooms and their associated details to the table in lines 77-89

**Comment by iranroman** [comment]
@adrianSRoman can you please have a look at the git history in this PR? For some reason all commits already in main are showing up here. Thanks.

**Comment by iranroman** [comment]
I have run inference with a pre-trained SELDnet23 model on 20 soundscapes generated with the `example_generation.py` script on the new rooms (which are completely unseen to the model), obtaining the following results: 

DAGA
```
SED metrics: Error rate: 0.69 [0.66, 0.73], F-score: 26.7 [21.69, 32.46]
DOA metrics: Localization error: 36.5 [30.17 , 44.09], Localization Recall: 45.5 [40.81, 50.91]
```

RSoANU
```
SELD score (early stopping metric): 0.52 [0.50, 0.56]
SED metrics: Error rate: 0.73 [0.69, 0.77], F-score: 22.2 [17.20, 26.02]
DOA metrics: Localization error: 33.3 [30.06 , 37.13], Localization Recall: 60.3 [54.30, 64.41]
```

MOTUS
```
TODO
```

**Comment by iranroman** [comment]
This PR is getting in shape. 

It would be great to get a code review from two more people:
@ChrisIck 
@adrianSRoman 
@sivannavis 
@abreuwallace 
@giovana-morais 
@sripathisridhar 

Thank you in advance. It would be great if we can merge over the weekend.

File: scripts/prepare_rirs.py [hunk 2]
```diff
@@ -274,33 +432,106 @@ def prepare_arni(path_raw, path_sofa, formats=["mic", "foa"]):
...
+        sofa = netCDF4.Dataset(sofa_file, "r")
+        irs = sofa.variables["Data.IR"][:]
+        orig_sr = sofa.variables["Data.SamplingRate"][:][0]
+        sofa.close()
+
+        for fmt in audio_fmts:
+            if fmt == "mic":
+                processed_irs = irs[:, __TETRA_CHANS_IN_EM32__, 0].T
+            else:
+                # place array2sh conversion here
```
Comment (giovana-morais) [inline scripts/prepare_rirs.py:462]: is this `array2sh` done? just making sure that you don't forget to add it here in case it is. hehe

**Review by giovana-morais — APPROVED** [review]
looks good to me!

File: scripts/prepare_rirs.py (hunk 2 above)
Comment (iranroman) [reply scripts/prepare_rirs.py:462]: thank you! we are working on it, but it won't be part of this PR yet.

**Comment by karinnacvijanovich** [comment]
A PR with fixed commit history is now open at PR #71. Closing this one.

**Pull request closed without merging**